app/models/creator.rb

```diff
@@ -6,14 +6,10 @@ class Creator < ApplicationRecord
   after_save :reindex_associated_works
 
   def reindex_associated_works
-    Publication.where(creator_id_ssim: id.to_s).find_each do |record|
-      record.update_index
-    end
-    Dataset.where(creator_id_ssim: id.to_s).find_each do |record|
-      record.update_index
-    end
-    ConferenceProceeding.where(creator_id_ssim: id.to_s).find_each do |record|
-      record.update_index
+    solr = Blacklight.default_index.connection
+    response = solr.get 'select', params: { q: "creator_id_ssim:#{id}" }
+    response['response']['docs'].each do |doc|
+      doc['has_model_ssim'].first.constantize.find(doc['id']).update_index
```

Comment:
I think you can just `ActiveFedora::Base.find(doc['id'].update_index` which might be a little clearer about what's going on here instead of trying to find the work class and turning it into a constant to call.

Reply:
Actually, you should be able to just call `CypripediumWork.find(doc['id']).update_index`.  I feel like that makes it a little clearer about what's going on here.